svgdigitizer/svgdigitizer.py

```diff
@@ -183,7 +191,44 @@ def parse_pathstring(self, path_string):
             posxy.append([x0, y0])
 
         return np.array(posxy)
-    
+
+
+    def sample_path(self, path_string):
+        '''samples a path with equidistant x segment by segment'''
+        path = Path(path_string)
+        xmin, xmax, ymin, ymax = path.bbox()
+        x_samples = np.linspace(xmin, xmax, int(abs(xmin - xmax)/self.transformed_sampling_interval))
+        points = []
+        for segment in path:
+            segment_path = Path(segment)
+            xmin_segment, xmax_segment, _, _ = segment.bbox()
+            segment_points = [[],[]]
+
+            for x in x_samples:
+                # only sample the x within the segment
+                if x >= xmin_segment and x <= xmax_segment:
+                    intersects = Path(Line(complex(x,ymin),complex(x,ymax))).intersect(segment_path)
+                    # it is possible that a segment includes both scan directions
+                    # which leads to two intersections
+                    for i in range(len(intersects)):
+                        point = intersects[i][0][1].point(intersects[i][0][0])
+                        segment_points[i].append((point.real, point.imag))   
+
+            # second intersection is appended in reverse order!! 
+            if len(segment_points[1]) > 0:    
+                segment_points[0].extend(segment_points[1][::-1]) 
+            # sometimes segments are shorter than sampling interval   
+            if len(segment_points[0]) > 0:    
+                first_segment_point = (segment.point(0).real, segment.point(0).imag)    
+
+                if (((segment_points[0][-1][0]-first_segment_point[0])**2+(segment_points[0][-1][1]-first_segment_point[1])**2)**0.5 >    
```

Comment:
I think there is no need for a square root here.

Reply:
I'm not sure if there are conditions where we need Pythagoras.